models/analytics/bridge_dim.sql

```diff
@@ -0,0 +1,35 @@
+{#
+Object Name: bridge_dim
+Purpose: To connect the location_dim to subject_id
+#}
+
+{{
+    config(
+        materialized='table'
+    )
+}}
+
+WITH extract_raw_data AS (
+    select 
+        sub.id AS subjects_id, 
+        sub.location AS location_json,
+        json_extract_path_text(sub.location::json, 'Ward') AS ward_name,
+        json_extract_path_text(sub.location::json, 'Block') AS block_name,
+        json_extract_path_text(sub.location::json, 'District') AS district_name,
+        json_extract_path_text(sub.location::json, 'Gram Panchayat') AS gp_name
+    from {{ ref ('subjects_cdc') }} AS sub
+)
+
+, transform_keys AS (
+    SELECT 
+    subjects_id,
+    CAST( {{ dbt_utils.generate_surrogate_key(['ward_name', 'block_name', 'district_name', 'gp_name']) }} AS VARCHAR) AS location_id
```

Comment:
Instead of creating another surrogate key here can't we just use the `location_dim_id` from the `location_dim` table itself? joining on the 4 location attributes

transform_keys AS (
    SELECT 
        erd.subject_id,
        ld.location_dim_id
    FROM extract_raw_data erd
    JOIN {{ ref('location_dim') }} ld
    ON erd.ward_name = ld.ward
    AND erd.block_name = ld.block
    AND erd.district_name = ld.district
    AND erd.gp_name = ld.gram_panchayat)